mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/validator/ParameterModelValidator.java

```diff
@@ -107,6 +101,40 @@ private void validateParameter(ParameterModel parameterModel, String ownerName,
 
     }
 
+    private void validateDefaultValue(ParameterModel parameterModel, String ownerName, String ownerModelType) {
+      if (parameterModel.isOverrideFromConfig() &&
+          !ownerModelType.equals(CONFIGURATION) && !ownerModelType.equals(CONNECTION_PROVIDER)) {
+        // We skip failing for configs and connection here since a different error is thrown in their own validators
+        addError(parameterModel,
+                 "Parameter '%s' in the %s '%s' is declared as a config override,"
+                     + " and must not provide a default value since one is already provided by the value"
+                     + " declared in the config parameter",
+                 parameterModel.getName(), ownerModelType, ownerName);
+      } else if (parameterModel.isRequired()) {
+        addError(parameterModel,
+                 "Parameter '%s' in the %s '%s' is required, and must not provide a default value",
+                 parameterModel.getName(), ownerModelType, ownerName);
+      }
+
+      if (!hasExpressionDefaultValue(parameterModel)) {
+        parameterModel.getType().accept(new MetadataTypeVisitor() {
+
+          @Override
+          public void visitString(StringType type) {
+            super.visitString(type);
+            type.getAnnotation(EnumAnnotation.class).ifPresent(enumAnnotation -> {
+              List<String> values = Stream.of(enumAnnotation.getValues()).map(Object::toString).collect(toList());
+              if (!values.contains(parameterModel.getDefaultValue().toString())) {
+                addError(parameterModel,
+                         "Parameter '%s' in the %s '%s' has a default value which is not listed as an available option",
```

Comment:
IMO this error message should include the default value that was set, and the list of values that the enum can take.